organizations.go

```diff
@@ -101,13 +105,23 @@ func (s *OrganizationServiceOp) List(opts *ListOptions) (orgs []Organization, re
 	}
 }
 
+func (s *OrganizationServiceOp) DefaultIncludes() []string {
+	return []string{}
+}
+
+func (s *OrganizationServiceOp) Hydrate(resource href.Hrefer, opts *GetOptions) (*Response, error) {
+	opts.Including(s.DefaultIncludes()...)
+	apiPathQuery := opts.WithQuery(resource.GetHref())
+
+	return s.client.DoRequest(http.MethodGet, apiPathQuery, nil, resource)
+}
+
 // Get returns a organization by id
 func (s *OrganizationServiceOp) Get(organizationID string, opts *GetOptions) (*Organization, *Response, error) {
-	endpointPath := path.Join(organizationBasePath, organizationID)
-	apiPathQuery := opts.WithQuery(endpointPath)
 	organization := new(Organization)
-
-	resp, err := s.client.DoRequest("GET", apiPathQuery, nil, organization)
+	href := path.Join(organizationBasePath, organizationID)
```

Comment:
If we implement a `GetBasePath() string` for `*ServiceOp`, and `NewResource() interface{} // return new(Organization)`, we could implement `Get` in a ServiceOp type embedded in `*ServiceOp`.

We would get `Get` and `Hydrate` for free in each endpoint (that uses Href and ID).

Reply:
Perhaps we could get Delete and others for free?

Reply:
It would be cool to have Get method for free. Just rarely there are still some resources with more than one ID param to lookup. not fitting the `Get(string, *GetOptions)` signature. Simple command ` grep -P "^\tGet\(" *go | grep string` shows me only vlan assigments are the case at the moment (because I have the #302 PR checked out). But I think API keys are special too.

Then, I'm not sure how/where we'd ensure that the Get method returns the desired resource type. Right now the type is declared in the Get signature `Get(DeviceID string, opts *GetOptions) (>>*Device<<, *Response, error)`. 

So I think, for the parent interface would have to implement the Hydrate method as you wrote it, and for every concrete ServiceOp we would have to implement:

* `func GetResourcePath(id string) string { return path.Join(deviceBasePath, id) }`
* `func DefaultIncludes() []string { return []string{"facility", "hardware_reservation", "project"} }` (this could be an attribute)
* `func NewResource() *Device { return new(Device) }`
* Get like
  ```golang
  // this could be copy-pasted or template-generated to all the service opts
  func (s *DeviceServiceOp) Get(id string, opts *GetOptions) (*Device, *Response, error) {
    resource := s.NewResource(s.GetResourcePath(id))        
    opts = opts.Including(s.DefaultIncludes())
    resp, err := s.Hydrate(resource, opts)
    if err != nil {
      resource = nil
    }
    return resource, resp, err
  }
  ```